### PR TITLE
feat: refactor the confidential compute endpoints

### PR DESCRIPTION
**Pull request by jorgeantonio21**

### CODE REVIEW

File: atoma-state/src/state_manager.rs [hunk 1]
```diff
@@ -1587,6 +1639,63 @@ impl AtomaState {
...
+        skip_all,
+        fields(
+            %stack_small_id,
+            %available_compute_units
+        )
+    )]
+    pub async fn lock_compute_units_for_stack(
+        &self,
+        stack_small_id: i64,
+        available_compute_units: i64,
```
Comment (Cifko) [inline atoma-state/src/state_manager.rs:1686]: maybe rename this to lock_compute_units

File: atoma-state/src/state_manager.rs [hunk 2]
```diff
@@ -1587,6 +1639,63 @@ impl AtomaState {
...
+    ) -> Result<()> {
+        let result = sqlx::query("UPDATE stacks SET already_computed_units = already_computed_units + $1 WHERE stack_small_id = $2")
```
Comment (Cifko) [inline atoma-state/src/state_manager.rs:1688]: you are allowing to set `already_computed_units` above the `num_compute_units` of this stack

**Review by Cifko — CHANGES_REQUESTED** [review]

File: atoma-state/src/state_manager.rs (hunk 2 above)
Comment (jorgeantonio21) [reply atoma-state/src/state_manager.rs:1688]: Nice catch !

File: atoma-state/src/state_manager.rs (hunk 1 above)
Comment (Cifko) [reply atoma-state/src/state_manager.rs:1687]: you didn't renamed it :-)

**Review by Cifko — APPROVED** [review]

File: atoma-state/src/state_manager.rs (hunk 2 above)
Comment (Cifko) [reply atoma-state/src/state_manager.rs]: push changes

**Pull request merged**